python/ray/tune/schedulers/resource_changing_scheduler.py

```diff
@@ -0,0 +1,222 @@
+import logging
+from typing import Dict, Optional, Union, Callable
+
+import pickle
+import warnings
+
+from ray.tune import trial_runner
+from ray.tune.resources import Resources
+from ray.tune.schedulers.trial_scheduler import FIFOScheduler, TrialScheduler
+from ray.tune.trial import Trial
+from ray.tune.utils.placement_groups import PlacementGroupFactory
+
+logger = logging.getLogger(__name__)
+
+
+class ResourceChangingScheduler(TrialScheduler):
+    """A utility scheduler to dynamically change resources of live trials.
+
+    Experimental. API may change in future releases.
+
+    The ResourceChangingScheduler works by wrapping around any other
+    scheduler and adjusting the resource requirements of live trials
+    in response to the decisions of the wrapped scheduler
+    through a user-specified `resources_allocation_function`. An example
```

Comment:
Since this is rendered with Sphinx, we'll need to use double backticks here (and in the other occurences)
```suggestion
    through a user-specified ``resources_allocation_function``. An example
```